sktime/classification/deep_learning/cnn.py

```diff
@@ -134,7 +143,11 @@ def _fit(self, X, y):
 
         check_random_state(self.random_state)
         self.input_shape = X.shape[1:]
-        self.model_ = self.build_model(self.input_shape, self.n_classes_)
+        self.model_ = (
```

Comment:
can you kindly explain this change? Should `self.model_` not always be `None`, and replaced even if it is already fitted?

Reply:
This is related to the `load` functionality. The thought here is that:
- user first creates object of the estimator class
- user then calls the `load` function with path
- `BaseDeepClassifier` loads the model into `self.model_` 
- user calls `model.fit`
- Within `CNNClassifier.fit` function, we first check if `self.model_` is initially defined (that is, if user has already loaded in the network) or it is `None`. If it is `None`, then we build the network.

Reply:
sounds like an odd workflow, can you explain why that would be desired or intuitive?

The "normal" save/load workflow is as follows:

A. save
A1. user builds model `model`
A2. user saves model to a location `saved_model`, via `model.save`

B. load
B1. user loads `model` from `saved_model`, via `load` (loose function)

Your workflow part B takes three steps.

We should also distinguish, are you trying to save/load a model *configuration* (unfitted), or a model *serialization* (fitted)?

Reply:
I was thinking in terms of saving/loading a fitted model (model serialisation). Are there any use cases where there would be a point of saving a model configuration?